src/treetensornetworks/projttns/abstractprojttn.jl

```diff
@@ -1,17 +1,22 @@
 abstract type AbstractProjTTN{V} end
 
+#accessors for fields common to all concrete subtypes
+environments(p::AbstractProjTTN) = p.environments
+operator(p::AbstractProjTTN) = p.operator
+underlying_graph(P::AbstractProjTTN) = underlying_graph(operator(P))
+pos(P::AbstractProjTTN) = P.pos
```

Comment:
These should only be defined for the concrete `ProjTTN` type, I don't think they will make sense for general `AbstractProjTTN` which could have very different data structures.

Reply:
Ok. Should `environments`, `operator`, and `pos` exist for `AbstractProjTTN` as error("Not Implemented"), so the developer knows that these accessors should be defined for any concrete type (regardless of what the field names actually are in the concrete types?).
Also in that case, I'd say
`underlying_graph(P::AbstractProjTTN) = underlying_graph(operator(P))`
should be defined only for AbstractProjTTN, unless a concrete type has reason to overwrite it?

Reply:
Yeah, they should error as you say.

I'm not sure if `underlying_graph(P::AbstractProjTTN) = underlying_graph(operator(P))` makes sense as a generic definition. For example, if it is a sum of operators, what would `operator` return?

Reply:
Ok, I'll treat it like the others.